jbi/steps.py

```diff
@@ -234,6 +234,8 @@ def maybe_update_components(context: ActionContext, parameters: ActionParams):
     candidate_components = set(parameters.jira_components)
     if context.bug.component:
         candidate_components.add(context.bug.component)
+    if context.bug.product_component:
+        candidate_components.add(context.bug.product_component)
```

Comment:
I feel like we have a few options when it comes to how we sync a bug's component to an issue. For example take a bug with a product `Product` and a component `General`, as you used in tests below. We could:
- Add the component without the prefix: `General` (what we were doing before this PR)
- Add the component with the product prefix: `Product::General`
- Add both the prefixed and non-prefixed value (what we're doing in this PR)
  - `General`
  - `Product::General`
- Add two components
  - `Product`
  - `General`

Should we give users a choice for how they'd like the component to appear on their issue? If we make our `candidate_components` too "wide" we run the risk of emitting lots of warnings that a component wasn't available in a project. But, an user may only care about the component being added to the issue in a certain way. For example, they may only set up the prefixed notation as valid components on their Jira project, so we'd get an error when trying to set the non-prefixed version.

Reply:
Very good point. We could have a parameter:

```md
* `use_bugzilla_product_as_jira_component`: `no` / `yes` / `prefix` / `both`
   For a Bugzilla ticket in `Core::General`, use `no` to set component as `General`, use `yes` to set components as `Core` and `General`, use `prefix` to set component as `Core::General`, and `both` to set components as `General` and `Core::General`
```

Reply:
Yep, some option in the config to control this would be good. I was also thinking something like this?

```yaml
  parameters:
    ...
    jira_components:
      sync_bug_component: <bool>
      sync_product_as_component: <bool>
      sync_bug_component_with_product_prefix: <bool>
      set_custom_components:
      - "my_custom_components"
```

Reply:
😎 niice
good idea to use a sub-object